App.js

```diff
@@ -316,7 +316,8 @@ export default class App extends React.Component {
                 <Stack.Screen
                   name="Chat"
                   component={ChatScreen}
-                  options={{headerShown: false}} />
+                  options={{headerShown: false}} 
+                  getId={({ params }) => params.id}/>
```

Comment:
What problem is this solving again?

Reply:
If we use `push` it would create duplicate entries for the same chat report. And if we use `navigate` it would just add a single chat report screen and just update the param i.e ID. So through this, we will be able to push different chat reports because now id is used to know if we need to push another chat screen or not(whereas previously we just relied on the screen name i.e Chat).

Before
-----
<img width="1530" alt="Screenshot 2022-12-08 at 5 08 21 PM" src="https://user-images.githubusercontent.com/35863227/206543472-a0442bfc-a511-4146-a18f-55dda9722a5f.png">

After
-----
<img width="1253" alt="Screenshot 2022-12-08 at 5 06 20 PM" src="https://user-images.githubusercontent.com/35863227/206543543-9f6a2151-98d0-407e-910f-f7e81fcc93f1.png">

More about it here: https://reactnavigation.org/docs/screen/#getid

This should solve the above problem that you described.

Reply:
>If we use push it would create duplicate entries for the same chat report

I'm having trouble understanding why this is a problem. I think if we are already on that page then it makes sense to not push a duplicate. But if it exists somewhere else in the stack then there's no reason why we can't push a duplicate entry? And actually seems like what I would expect to happen.

@mountiny what do you think about using `getId()` here?

Reply:
I see your concern here so the way to tackle this will be just to check during push if we are on the same page and not push duplicate entry.

Reply:
It took me a bit to realize what the issue is because I thought the navigator would treat `chat/1` and `chat/2` as different routes but seems like it does not. Hence that is why we need the `getID` because otherwise navigate will not add the different chats to the history.

I tested this and basically you end up moving between the chats but not being able to go back so this change seems required to me

Reply:
>I see your concern here so the way to tackle this will be just to check during push if we are on the same page and not push duplicate entry.

Yeah is there some easy way we can do this? I think it will be rare that we would be navigating to a link from a chat to the same chat.

>It took me a bit to realize what the issue is because I thought the navigator would treat chat/1 and chat/2 as different routes but seems like it does not. 

Sorry I didn't understand this explanation. What do you mean by "as different routes"?

>Hence that is why we need the getID because otherwise navigate will not add the different chats to the history.

Right - but if we use `push()` then it will always add one to the history?